banyand/property/shard.go

```diff
@@ -225,3 +232,45 @@ func (s *shard) search(ctx context.Context, indexQuery index.Query, limit int,
 	}
 	return data, nil
 }
+
+func (s *shard) prepareForMerge(src []*roaring.Bitmap, segments []segment.Segment, _ uint64) (dest []*roaring.Bitmap, err error) {
+	if len(segments) == 0 || len(src) == 0 || len(segments) != len(src) {
+		return src, nil
+	}
+	for segID, seg := range segments {
+		var docID uint64
+		for ; docID < seg.Count(); docID++ {
+			hasDeleted := false
+			var sourceData []byte
+			_ = seg.VisitStoredFields(docID, func(field string, value []byte) bool {
+				if field == deleteField {
+					hasDeleted = convert.BytesToBool(value)
```

Comment:
Please include the deletion time for this field. By doing so, there’s no need to parse the source field.